spec/support/shared_entity_specs.rb

```diff
@@ -70,3 +70,19 @@ def validate_values(value, parsed_value, type)
     end
   end
 end
+
+shared_examples "a relayable Entity" do
+  let(:instance) { described_class.new(data) }
+
+  context "signatures generation" do
+    it "computes correct signatures for the entity" do
+      xml = DiasporaFederation::Salmon::XmlPayload.pack(instance)
+      hash = DiasporaFederation::Salmon::XmlPayload.hash(xml.at_xpath("post/*[1]"))
```

Comment:
you can use `instance.to_h` and `xml.at_xpath("post/*[1]/author_signature").text` here, so you don't need the `XmlPayload.hash`

And you should use a `data` with empty signatures, so you can check if `pack` updates it. If they are set before, this test is useless.